pkg/controller/flink/task_manager_controller.go

```diff
@@ -20,11 +20,12 @@ import (
 )
 
 const (
-	TaskManagerNameFormat     = "%s-%s-tm"
-	TaskManagerPodNameFormat  = "%s-%s-tm-pod"
-	TaskManagerContainerName  = "taskmanager"
-	TaskManagerArg            = "taskmanager"
-	TaskManagerHostnameEnvVar = "TASKMANAGER_HOSTNAME"
+	TaskManagerNameFormat           = "%s-%s-tm"
+	TaskManagerPodNameFormat        = "%s-%s-tm-pod"
+	TaskManagerVersionPodNameFormat = "%s-%s-tm-%s-pod"
```

Comment:
Is it necessary to put version in the pod name? I think they'll still be distinguishable by deploy hash.

Reply:
I was thinking more along the lines of how we could use the pod name for distinguishing other external things (like logs in Lyft's case) that would benefit from regex like searches.